src/java/org/apache/cassandra/db/commitlog/AbstractCommitLogSegmentManager.java

```diff
@@ -111,12 +111,20 @@ void start()
     {
         // For encrypted segments we want to keep the compression buffers on-heap as we need those bytes for encryption,
         // and we want to avoid copying from off-heap (compression buffer) to on-heap encryption APIs
-        BufferType bufferType = commitLog.configuration.useEncryption() || !commitLog.configuration.useCompression()
-                                ? BufferType.ON_HEAP
-                                : commitLog.configuration.getCompressor().preferredBufferType();
+        BufferType bufferType = commitLog.configuration.isDirectIOEnabled()
```

Comment:
wow to read this is quite a challenge! Could you rewrite this somehow with if / else?